macro/README.md

```diff
@@ -0,0 +1,64 @@
+# Helper macros for Host development 
+
+This crate provides a few helper macros for Host development.
+
+## Usage
+
+Add this to your `Cargo.toml`:
+
+```toml
+[dependencies]
+nexus-macro = { path = "../macro" }
+nexus-profiler = { path = "../macro/profiler" }
+```
+
+## Profile macros
+
+### Requirements
+
+Install `go` programming language.
+
+#### MacOS
+
+```bash
+brew install go
+```
+
+#### Linux
+
+```bash
+sudo apt install golang-go
+```
+
+### How to use `#[profile]` macro
+
+At any host function, you can use `#[profile]` to profile the function.
+
+```rust
+use nexus_macro::profile;
+
+#[profile]
+pub fn eval_inst(vm: &mut NexusVM<impl Memory>) -> Result<()> {
+    /// .....
+    /// .....
+}
+
+#[profile("is_satisfied.pb")]
+pub fn is_satisfied<C: CommitmentScheme<G>>(
+        &self,
+        U: &R1CSInstance<G, C>,
+        W: &R1CSWitness<G>,
+        pp: &C::PP,
+    ) -> Result<(), Error> {
+        /// ...
+        /// ....
+    }
+
+```
+
+When the profile output name is undefined, the function name is used as default with suffix `.pb`.
+The first macro will write the profile data to `eval_inst.pb`. The second macro will write the profile data to `is_satisfied.pb`.
```

Comment:
What happens if `is_satisfied` is compiled multiple times on different `CommitmentScheme`s?

Is it allowed to use the same file name for profile data from multiple functions?

Reply:
Thanks for the question @yoichi-nexus , this macro is intended to be used in a one-time call function. 
I tried adding the current time to the filename, however, the filename is either become clustered with nanosecond number or overlapped each other. 

It won't gather profile data if a function is called repetitively.

Reply:
> used in a one-time call function

That's new to me. It's worth documenting if not written somewhere already.

Reply:
Added in https://github.com/nexus-xyz/nexus-zkvm/pull/261/commits/c56058a796d40a4c4c4ed4cd4d1756bbf726fc0f